Blockzilla/SearchHistoryUtils.swift

```diff
@@ -17,55 +17,37 @@ struct textSearched {
         guard let text = dictionary["text"], let isCurrentSearch = dictionary["isCurrentSearch"] else { return nil }
         self.init(text: text as! String, isCurrentSearch: isCurrentSearch as! Bool)
     }
-
-    var propertyListRepresentation: [String: Any] {
-        return ["text": text, "isCurrentSearch": isCurrentSearch]
-    }
-
 }
 
 class SearchHistoryUtils {
 
     static var isFromURLBar = false
     static var isNavigating = false
     static var isReload = false
+    private static var currentStack = [textSearched]()
 
     static func pushSearchToStack(with searchedText: String) {
-        var currentStack = [textSearched]()
-
-        if let propertylistSearchesRead = UserDefaults.standard.object(forKey: "searchedHistory") as? [[String: Any]] {
-            currentStack = propertylistSearchesRead.compactMap { textSearched(dictionary: $0) }
-
-			// Check whether the lastSearch is the current search. If not, remove subsequent searches
-			if let lastSearch = currentStack.last, !lastSearch.isCurrentSearch {
-				for index in 0..<currentStack.count {
-					if currentStack[index].isCurrentSearch {
-						currentStack.removeSubrange(index+1..<currentStack.count)
-						break
-					}
-				}
-			}
-
+        // Check whether the lastSearch is the current search. If not, remove subsequent searches
+        if let lastSearch = currentStack.last, !lastSearch.isCurrentSearch {
             for index in 0..<currentStack.count {
-                currentStack[index].isCurrentSearch = false
+                if currentStack[index].isCurrentSearch {
```

Comment:
You should probably use the `currentStack[safe: index]` syntax to be safe. Its the same as regular index-based accessors, except it returns an optional value.

Reply:
Disregard my other comment here. I feel like there's probably a drastically simpler way to solve the problem this class aims to solve, but it might be safer to just leave the code in-tact and simply replace the backing store with an in-memory array as you did.